Source/WebCore/html/HTMLFormElement.cpp

```diff
@@ -295,6 +295,15 @@ void HTMLFormElement::submitIfPossible(Event* event, HTMLFormControlElement* sub
     m_isSubmittingOrPreparingForSubmission = true;
     m_shouldSubmit = false;
 
+    bool processingUserGesture = !submitter;
```

Comment:
We have other ways to know if we're in user gesture. Does this match the behavior of the kind of "in user gesture" used for permissions? Should it? If not, maybe use a different name.

Reply:
It was already passed in with that value in `HTMLFormElement::submit`, which uses that variable name, so I assume it's correct:

```
HTMLFormElement::submit(Event* event, bool processingUserGesture, FormSubmissionTrigger trigger, HTMLFormControlElement* submitter)
```

Reply:
Assuming that the HTMLFormControlElement means "in a user gesture" doesn't seem very future-proof. Can we just call `UserGestureIndicator::processingUserGesture()` ?